File: restaurant-app/lib/src/features/printers/models/restaurant_printer.dart
class RestaurantPrinter {
  final String id;
  final String name;
  final String type; // general | caisse | cuisine | bar
  final String ip;
  final int port;
  final bool isEnabled;
  final int paperWidthMm; // 58 or 80
  final String connectionType; // network

  const RestaurantPrinter({
    required this.id,
    required this.name,
    required this.type,
    required this.ip,
    required this.port,
    required this.isEnabled,
    required this.paperWidthMm,
    this.connectionType = 'network',
  });

  factory RestaurantPrinter.fromMap(Map<String, dynamic> map) {
    return RestaurantPrinter(
      id: map['id']?.toString() ?? '',
      name: map['name']?.toString() ?? '',
      type: map['type']?.toString() ?? 'general',
      ip: map['ip']?.toString() ?? '',
      port: (map['port'] as num?)?.toInt() ?? 9100,
      isEnabled: (map['isEnabled'] as bool?) ??
          (map['is_enabled'] as bool?) ??
          true,
      paperWidthMm: (map['paperWidthMm'] as num?)?.toInt() ??
          (map['paper_width_mm'] as num?)?.toInt() ??
          80,
      connectionType: map['connectionType']?.toString() ??
          map['connection_type']?.toString() ??
          'network',
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'type': type,
      'ip': ip,
      'port': port,
      'isEnabled': isEnabled,
      'paperWidthMm': paperWidthMm,
      'connectionType': connectionType,
    };
  }

  RestaurantPrinter copyWith({
    String? id,
    String? name,
    String? type,
    String? ip,
    int? port,
    bool? isEnabled,
    int? paperWidthMm,
    String? connectionType,
  }) {
    return RestaurantPrinter(
      id: id ?? this.id,
      name: name ?? this.name,
      type: type ?? this.type,
      ip: ip ?? this.ip,
      port: port ?? this.port,
      isEnabled: isEnabled ?? this.isEnabled,
      paperWidthMm: paperWidthMm ?? this.paperWidthMm,
      connectionType: connectionType ?? this.connectionType,
    );
  }
}

File: backend/src/models/RestaurantPrinter.js
import { DataTypes } from "sequelize";
import { sequelize } from "../config/database.js";

const RestaurantPrinter = sequelize.define(
  "RestaurantPrinter",
  {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true,
    },
    restaurant_id: {
      type: DataTypes.UUID,
      allowNull: false,
      references: { model: "restaurants", key: "id" },
      onDelete: "CASCADE",
      onUpdate: "CASCADE",
    },
    name: {
      type: DataTypes.STRING(255),
      allowNull: false,
      comment: "Nom affiché: Caisse 1, Cuisine, Bar...",
    },
    type: {
      type: DataTypes.STRING(50),
      allowNull: false,
      defaultValue: "general",
      comment: "general | caisse | cuisine",
    },
    ip: {
      type: DataTypes.STRING(45),
      allowNull: false,
      comment: "Adresse IP de l'imprimante réseau (LAN)",
    },
    port: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 9100,
      comment: "Port RAW (souvent 9100) pour ESC/POS",
    },
    is_enabled: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: true,
    },
    paper_width_mm: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 80,
      comment: "58 ou 80 mm",
    },
  },
  {
    tableName: "restaurant_printers",
    timestamps: true,
    underscored: true,
    createdAt: "created_at",
    updatedAt: "updated_at",
  }
);

export default RestaurantPrinter;
